ibm_mq/datadog_checks/ibm_mq/collectors/queue_metric_collector.py

```diff
@@ -202,17 +202,30 @@ def get_pcf_queue_status_metrics(self, queue_manager, queue_name, tags):
             # Response is a list. It likely has only one member in it.
             for queue_info in response:
                 for mname, values in iteritems(metrics.pcf_metrics()):
-                    failure_value = values['failure']
-                    pymqi_value = values['pymqi_value']
-                    mname = '{}.queue.{}'.format(metrics.METRIC_PREFIX, mname)
-                    m = int(queue_info[pymqi_value])
-
-                    if m > failure_value:
-                        self.send_metric(GAUGE, mname, m, tags=tags)
-                    else:
-                        msg = "Unable to get {}, turn on queue level monitoring to access these metrics for {}"
-                        msg = msg.format(mname, queue_name)
-                        self.log.debug(msg)
+                    metric_name = '{}.queue.{}'.format(metrics.METRIC_PREFIX, mname)
+                    try:
+                        if callable(values):
+                            metric_value = values(queue_info)
+                            if metric_value is not None:
+                                self.send_metric(GAUGE, metric_name, metric_value, tags=tags)
+                            else:
+                                msg = "Unable to get {}, turn on queue level monitoring to access these metrics for {}"
```

Comment:
Could you make the error message more descriptive since there could be other reasons the metrics aren't reported, ie errors calculating the date, or the user could be using python 2

Reply:
Thanks, good point! I've added python 2 support and added additional messaging to check the queue status command.